server/src/main/java/org/apache/druid/segment/loading/SegmentLoaderConfig.java

```diff
@@ -50,7 +50,7 @@
   private int announceIntervalMillis = 0; // do not background announce
 
   @JsonProperty("numLoadingThreads")
-  private int numLoadingThreads = JvmUtils.getRuntimeInfo().getAvailableProcessors();
+  private int numLoadingThreads = Math.min(1, JvmUtils.getRuntimeInfo().getAvailableProcessors() / 6);
```

Comment:
Oops, I think this should be `Math.max` so the value isn't always 0 or 1?

Reply:
whoops , thanks for catching this!